tests/unit/test_hardware.py

```diff
@@ -328,3 +328,18 @@ def test_parse_maximal_constraint() -> None:
     print(textwrap.dedent(hw_spec_out))
 
     assert tmt.utils.dict_to_yaml(hw.constraint.to_spec()) == textwrap.dedent(hw_spec_out).lstrip()
+
+    def test_parse_or_constraint() -> None:
```

Comment:
It is a useful test, let's keep it, but it's not testing the fix: parsing of `or` was correct, the problem was in how the Beaker plugin translated it into XML. To test that, we should have a test in `tests/unit/provision/mrack/test_hw.py`, calling `constraint_to_beaker_filter()` and checking its return value's `to_mrack()` gives the expected output. Something that we already have for various constraint-specific translations, this one would need to call the whole `constraint_to_beaker_filter()`. And we could have some nested constraints, to make sure the Beaker plugin handles them correctly, e.g.

```yaml
or:
  - hostname: == dummy1.redhat.com
  - or:
    - hostname: == dummy2.redhat.com
    - or:
      - hostname: == dummy3.redhat.com
      - hostname: == dummy4.redhat.com
```

And check what `constraint_to_beaker_filter()` returns :)

Reply:
Updated^^: https://github.com/teemtee/tmt/pull/3327/commits/6033f2c459046fe23b2e2de23a7b7aa2adf5846a